cardano-api/test/Test/Cardano/Api/Typed/CBOR.hs

```diff
@@ -155,38 +156,18 @@ prop_roundtrip_UpdateProposal_CBOR :: Property
 prop_roundtrip_UpdateProposal_CBOR =
   roundtrip_CBOR AsUpdateProposal genUpdateProposal
 
+prop_roundtrip_Tx_Cddl :: Property
+prop_roundtrip_Tx_Cddl = H.property $ do
+  AnyCardanoEra era <- H.forAll $ Gen.element [minBound..AnyCardanoEra AlonzoEra] --TODO: Babbage era
+  x <- forAll $ genTx era
+  H.tripping x serialiseTxLedgerCddl (deserialiseTxLedgerCddl era)
 
-test_roundtrip_Tx_Cddl :: [TestTree]
-test_roundtrip_Tx_Cddl =
-  [ testPropertyNamed (show era) (fromString (show era)) $ roundtrip_Tx_Cddl anyEra
-  | anyEra@(AnyCardanoEra era) <- [minBound..(AnyCardanoEra AlonzoEra)] --TODO: Babbage era
-  ]
-
-test_roundtrip_TxWitness_Cddl :: [TestTree]
-test_roundtrip_TxWitness_Cddl =
-  [ testPropertyNamed (show era) (fromString (show era)) $ roundtrip_TxWitness_Cddl era
-  | AnyCardanoEra era <- [minBound..(AnyCardanoEra AlonzoEra)] --TODO: Babbage era
-  , AnyCardanoEra era /= AnyCardanoEra ByronEra
-  ]
+prop_roundtrip_TxWitness_Cddl :: Property
+prop_roundtrip_TxWitness_Cddl = H.property $ do
+  AnyShelleyBasedEra era <- H.forAll $ Gen.element [AnyShelleyBasedEra ShelleyBasedEraShelley .. AnyShelleyBasedEra ShelleyBasedEraAlonzo]
```

Comment:
`AnyShelleyBasedEra` let's me generate Shelley based eras excluding Byron in a type-safe way.

It must not be `minBound` because `serialiseWitnessLedgerCddl` doesn't support Byron at the type-level.